cpp/Ice/customError/Chatbot.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "Chatbot.h"
+#include "../../common/Time.h"
+
+#include <Ice/Ice.h>
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+
+string
+Server::Chatbot::greet(string name, const Ice::Current&)
+{
+    cout << "Dispatching greet request { name = '" << name << "' }" << endl;
+
+    if (name.empty())
+    {
+        // ObjectNotExistException is a dispatch exception with a reply status of 'ObjectNotExist'.
+        throw Ice::ObjectNotExistException{__FILE__, __LINE__};
+    }
+    else if (name == "alice")
+    {
+        // This is a bogus error since there is no authentication in this demo.
```

Comment:
I would prefer if we use "dummy" instead of "bogus".

Reply:
I coped it from C#, should I change it there as well?

Reply:
Yes. I think we should.

Reply:
Make `fake`? I don't find dummy appropriate here.

Reply:
Fake is fine. Or we can also write something like:

```
/ Simulate an authentication error by throwing a dispatch exception with the Unauthorized error code.
// Note: This is a demo; no real authentication logic is implemented.
```